cmd/storage-datatypes.go

```diff
@@ -123,13 +123,17 @@ type FileInfo struct {
 	// Version of the file.
 	VersionID string
 
+	// Version of tier object delete marker
+	TierObjDelVersionID string
+
```

Comment:
Instead of keep adding all these fields, which are taking up memory and breaking compatibility each time why can't we keep all this as metadata?

If you want it to be neat to use, you can just add `func (f *FileInfo) TierObjDelVersionID() string` that fetches it from the metadata. We should be reducing the size of `FileInfo`, not adding to it. 

TBH everything that is stored as optional metadata should just be transferred like that IMO.

Reply:
@klauspost fair point. I shall move the new fields into metadata.

Reply:
@krisis Not a show stopper, but see what is feasible.